cli/src/main/java/org/jboss/as/cli/batch/impl/DefaultBatch.java

```diff
@@ -103,8 +106,25 @@ public ModelNode toRequest() {
         composite.get(Util.OPERATION).set(Util.COMPOSITE);
         composite.get(Util.ADDRESS).setEmptyList();
         final ModelNode steps = composite.get(Util.STEPS);
-        for(BatchedCommand cmd : commands) {
-            steps.add(cmd.getRequest());
+        for (BatchedCommand cmd : commands) {
+            CommandContext ctx = cmd.getCommandContext();
+            ModelNode request = cmd.getRequest();
+            if (ctx.getConfig().isValidateOperationRequests()) {
+                try {
+                    ctx.set(Scope.REQUEST, Util.DESCRIPTION_RESPONSE,
+                            cmd.getDescriptionResponse());
+                    ModelNode opDescOutcome = Util.validateRequest(ctx, request);
+                    if (opDescOutcome != null) { // operation has params that might need to be replaced
+                        Util.replaceFilePathsWithBytes(request, opDescOutcome);
```

Comment:
In this case, we have to do it. Because if the validation is disabled the substitution of file paths won't happen and the request will fail.
But as I wrote before, it's a separate issue and not a regression introduced by these changes.